frontend/src/citizen-frontend/messages/MessagesPage.tsx

```diff
@@ -58,6 +82,23 @@ export default React.memo(function MessagesPage() {
           }
         })}
       </StyledFlex>
+      {editorVisible && receivers.isSuccess && (
+        <MessageEditor
+          receiverOptions={receivers.value}
+          onSend={(message: SendMessageParams) =>
+            sendMessage(message)
+              .then(() => {
+                setEditorVisible(false)
+                setDisplaySendError(false)
+                refreshThreads()
+              })
+              .catch(() => setDisplaySendError(true))
```

Comment:
This `catch` is never reached because all exceptions are mapped to `Failure` in `sendMessage`. The failure state needs to be handled in `.then`